application/src/main/java/org/thingsboard/server/controller/DeviceController.java

```diff
@@ -438,7 +436,8 @@ public void onFailure(Throwable t) {
     @PreAuthorize("hasAnyAuthority('TENANT_ADMIN', 'CUSTOMER_USER')")
     @RequestMapping(value = "/customer/device/{deviceId}/claim", method = RequestMethod.DELETE)
     @ResponseStatus(value = HttpStatus.OK)
-    public DeferredResult<ResponseEntity> reClaimDevice(@PathVariable(DEVICE_ID) String strDeviceId) throws ThingsboardException {
+    public DeferredResult<ResponseEntity> reClaimDevice(@PathVariable(DEVICE_ID) String strDeviceId,
+                                                        @RequestParam(required = false) String secretKey) throws ThingsboardException {
```

Comment:
Let's make a key consist of only device id. The secret key will be used for comperison, but will be stored in the cache value and not the key.